tooling/nargo/src/ops/foreign_calls.rs

```diff
@@ -76,10 +126,92 @@ impl ForeignCall {
                     ],
                 })
             }
-            None => panic!("unexpected foreign call {foreign_call_name:?}"),
+            Some(ForeignCall::CreateMock) => {
+                let mock_oracle_name = Self::parse_string(&foreign_call.inputs[0]);
+                assert!(ForeignCall::lookup(&mock_oracle_name).is_none());
```

Comment:
This could be its own error as well, but like the other comment I am fine leaving this for another PR.